src/include/misc.h

```diff
@@ -333,19 +333,33 @@
 #define WT_STREQ(s, cs) (sizeof(cs) == 2 ? (s)[0] == (cs)[0] && (s)[1] == '\0' : strcmp(s, cs) == 0)
 
 /*
- * Check if a non-empty string matches a byte sequence of len bytes. The macro expects both strings
- * to be of non-zero length in order to be a match. The bytes argument does not need to be
- * null-terminated. Similar to other string related functions, this is not expected to work on NULL
- * values.
+ * Check if a literal string matches the length and content of the supplied bytes/len pair. The
+ * bytes argument does not need to be null-terminated, and it may be null if the supplied length is
+ * zero. Note this is different from the standard strncmp function, which with a supplied zero
+ * length, will always be true.
+ */
+#define WT_STRING_LIT_MATCH(str, bytes, len) \
+    ((len) == strlen("" str "") && strncmp(str, bytes, len) == 0)
+
+/*
+ * Identical to WT_STRING_LIT_MATCH, except that this works with non-literal strings. It is slightly
+ * slower, so WT_STRING_LIT_MATCH is always preferred for literal strings.
  */
 #define WT_STRING_MATCH(str, bytes, len) __wt_string_match(str, bytes, len)
```

Comment:
Is there any caller of `WT_STRING_MATCH` left? I didn't see any. I saw a bunch of `WT_CONFIG_MATCH` and everything else was literal. But it is definitely possible my eyes missed it.

Reply:
I think there are these:
```
meta/meta_ckpt.c:          WT_STRING_MATCH(blk_mod->id_str, blk->id_str, strlen(blk->id_str))) {
schema/schema_open.c:        match = idxname == NULL || WT_STRING_MATCH(name, idxname, len);
tiered/tiered_handle.c:        if (!WT_STRING_MATCH(key, object, strlen(object)))
txn/txn_ckpt.c:            if (WT_STRING_MATCH(ckpt->name, name, len)) {
txn/txn_ckpt.c:        if (!matched && !WT_STRING_MATCH(ckpt->name, name, len))
txn/txn_ckpt.c:        if (WT_STRING_MATCH(ckpt->name, name, len))
```

Reply:
Ah, that makes a lot of sense. Then ignore my suggestion to remove it!!